src/mock/account.js

```diff
@@ -115,10 +122,16 @@ export function genAccount(id: number | string): Account {
     name: rng.nextString(rng.nextInt(4, 34)),
     balanceByDay: {}
   };
+
+  let total = 0;
   account.operations = Array(operationsSize)
     .fill(null)
     .reduce(ops => {
       const op = genOperation(account, ops, currency, rng);
+      if (total + op.amount < 0) {
+        op.amount = -op.amount;
+      }
+      total += op.amount;
```

Comment:
how does it work? genOperation don't go back in time as i thought?

Reply:
Everytime before a new op is added to list, we ensure that the total balance will be positive. If not, we instead add the `-op.amount`.

Reply:
but genOperation goes back in time: https://github.com/meriadec/ledger-wallet-common/blob/8d899c190a5ec5149fadb72766c0ab36128dcedf/src/mock/account.js#L41-L51

so i'm not sure this works 🤔 balance needs to be calculated reversely

Reply:
let's discuss more tomorrow, but i'm not sure this works 🤔  maybe it does & i totally missed something the other day :o .